src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/AtlasDiffCommand.java

```diff
@@ -163,70 +181,27 @@ public int execute() // NOSONAR
-        final Optional<Change> changeOptional = diff.generateChange();
+        final AtlasDiffCommandContext context = new AtlasDiffCommandContext(beforeAtlasFile,
+                afterAtlasFile, useGeoJson, useLdGeoJson, fullText, selectedIdentifier,
+                selectedType, recursive);
 
-        if (changeOptional.isPresent())
+        if (beforeAtlasFile.isDirectory() && afterAtlasFile.isDirectory())
         {
-            Change change = changeOptional.get();
-            if (this.optionAndArgumentDelegate.hasOption(ID_OPTION_LONG)
-                    && this.optionAndArgumentDelegate.hasOption(TYPE_OPTION_LONG))
-            {
-                final Optional<FeatureChange> featureChangeOptional = change.changeFor(selectedType,
-                        selectedIdentifier);
-                if (featureChangeOptional.isPresent())
-                {
-                    change = new ChangeBuilder().add(featureChangeOptional.get()).get();
-                }
-                else
-                {
-                    this.outputDelegate.printlnWarnMessage(
-                            "No change found for " + selectedType + " " + selectedIdentifier);
-                    return 0;
-                }
-            }
-            final String serializedString;
-            if (useGeoJson)
-            {
-                serializedString = new ChangeGeoJsonSerializer().convert(change);
-            }
-            else if (useLdGeoJson)
-            {
-                serializedString = change.toLineDelimitedFeatureChanges();
-            }
-            else if (fullText)
-            {
-                final PrettifyStringFormat featureChangeFormat = this.optionAndArgumentDelegate
-                        .getOptionArgument(FEATURE_CHANGE_FORMAT_OPTION_LONG,
-                                PrettifyStringFormat::valueOf)
-                        .orElse(DEFAULT_PRETTY_FEATURE_CHANGE_FORMAT);
-                final PrettifyStringFormat completeEntityFormat = this.optionAndArgumentDelegate
-                        .getOptionArgument(COMPLETE_ENTITY_FORMAT_OPTION_LONG,
-                                PrettifyStringFormat::valueOf)
-                        .orElse(DEFAULT_PRETTY_COMPLETE_ENTITY_FORMAT);
-                serializedString = change.prettify(featureChangeFormat, completeEntityFormat, false)
-                        + "\n";
-            }
-            else
-            {
-                final StringBuilder builder = new StringBuilder();
-                change.changes().forEach(featureChange ->
-                {
-                    builder.append(featureChange.explain());
-                    builder.append("\n");
-                });
-                serializedString = builder.toString();
-            }
-            this.outputDelegate.printlnStdout(serializedString);
-            return 1;
+            final boolean result = this.compute(context, beforeAtlasFile, afterAtlasFile);
+            return result ? 1 : 0;
```

Comment:
Maybe just change `compute` to return an int?

Reply:
Changing it to return the number of diffs. Do you think I need to keep the command's output to 1 though? What if the command returns 465? is that an issue?

Reply:
You probably don't want to return `456`. Generally a program return value should be between 0-255, with ~128-255 being shell-reserved values. Most CLI diff tools follow the convention of `1` when a diff is detected and `0` (the OK value) when no diff is detected.